[AV-68571] Vector Search - The Capella Files

sarahlwelton opened the pull request:
This PR contains all of the documentation changes required for Vector Search on Capella. 

This PR will remain in draft until more review comments have been received. 

View the contents of the PR in context here: https://preview.docs-test.couchbase.com/sarah-AV-68571/cloud/vector-search/vector-search.html 

Username and password is the same as for all docs preview sites: https://couchbasecloud.atlassian.net/wiki/spaces/DOCS/pages/1971224949/Docs+Team+demo+site+passwords

sarahlwelton committed:
[AV-68976] Adjust all UI-based search index creation topics to match Figma designs.

sarahlwelton committed:
[AV-68976] Tweak explanation of mapping vs. type mapping

sarahlwelton committed:
[AV-68976] More edits to description of mappings/type mappings.

Adding missing context/property to search-index-params.adoc

sarahlwelton committed:
[AV-69144] Updating code samples and Search Indexes overview to reflect scoping changes

sarahlwelton committed:
[AV-69145] Adding topic import-search-index.adoc

Adding link from index-aliases.adoc and nav.adoc

sarahlwelton committed:
[AV-68976] Changing location of topics in nav.adoc

Changing title of quick-index-supported-languages.adoc

sarahlwelton committed:
Merge branch 'AV-68976-search-quick-editor' into AV-68571-vector-search-7-6-master

sarahlwelton committed:
Merge branch 'AV-69145-search-import-index' into AV-68571-vector-search-7-6-master

sarahlwelton committed:
[AV-68962] First draft of vector search overview

Created nav.adoc partial for new vector search module.

Updated search.adoc to provide brief overview of Vector Search and link.

sarahlwelton committed:
[AV-68962] Adding more notes about vector search.

sarahlwelton committed:
[AV-68962] Adding link to OpenAI documentation

sarahlwelton committed:
[AV-68962] More language tweaks on overview

sarahlwelton committed:
[AV-68962] Another language tweak

sarahlwelton committed:
[AV-68966] Initial draft of create and run vector search index topics for REST API

sarahlwelton committed:
[AV-68966] Fixing typo in include

sarahlwelton committed:
[AV-68571] Commenting out non-existent entry for now

sarahlwelton committed:
[AV-68962] Language tweak after discussion with SME

sarahlwelton committed:
Merge branch 'AV-68962-vector-search-overview' into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68976] Making changes to reflect current UI, instead of Figma designs.

Updating descriptions of type mappings and GeoJSON objects.

Fitting everything into nav.adoc

Updating links in some topics and changing anchors to support better understanding of type mappings and other changes.

sarahlwelton committed:
Merge branch 'AV-68976-search-quick-editor' into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68962] Copying over changes from Server version.

sarahlwelton committed:
Merge branch 'AV-68962-vector-search-overview' into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68963] Adding new vector settings to child-field-options-reference

Adding ip and vector type to field-data-types-reference

Making consistency changes and adding vector fields to quick-index-field-options

Adding page-toclevels, vector properties, and ip type to search-index-params.

Correcting page links in search-request-params.

Adding partial for vector search field descriptions

Adding topic for creating a vector search index with the UI

Adding UI topic back into the nav for vector search

sarahlwelton committed:
[AV-68964] Adding changes for running a vector search with the REST API, based on Server work

sarahlwelton committed:
[AV-68964] Adding changes for running a vector search with the UI, based on Server work

sarahlwelton committed:
Merge branch 'AV-68964-run-vector-search-ui' of https://github.com/couchbaselabs/docs-devex into AV-68964-run-vector-search-ui

sarahlwelton committed:
[AV-68967] Language and link fixes to newer Search docs, plus adding a Vector Search specific example

sarahlwelton committed:
[AV-68967] Some more link and text cleanup - some things were too Server specific

sarahlwelton committed:
[AV-68962] Adjusting note to link to relevant documentation and explain what happens

sarahlwelton committed:
Merge branch 'AV-68962-vector-search-overview' into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68976] Building out See Also section in customize-index

sarahlwelton committed:
Merge branch 'AV-68976-search-quick-editor' into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68966] Remove unnecessary "size" parameter

sarahlwelton committed:
Merge branch 'AV-68966-vector-search-rest-api' into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68571] Correcting incorrect table parameter

sarahlwelton committed:
[AV-68571] Removing unnecessary period in search.adoc

Fixing broken links in field-data-types-reference.adoc and customize-index.adoc

sarahlwelton committed:
[AV-68962] Adding link to integrations page and removing comments

sarahlwelton committed:
Merge branch 'capella' into AV-68571-vector-search-7-6-capella

dgnicholson commented on modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:
```diff
@@ -572,4 +614,6 @@ LIMIT 3;
 
-xref:server:fts:full-text-intro.adoc[Full Text Search]
+xref:search:search.adoc[]
+xref:vector-search:vector-search.adoc
```
Broken links / missing bullets: 

```suggestion
* xref:search:search.adoc[]
* xref:vector-search:vector-search.adoc[]
```

dgnicholson commented on modules/search/partials/vector-search-field-descriptions.adoc:
```diff
@@ -0,0 +1,37 @@
+Dot product similarity is commonly used by Large Language Models (LLMs).
+Use *dot_product* to get the best results with an embedding model that uses dot product similarity. 
+
+* *l2_norm*: Also known as euclidean distance. 
```
Super Tiny Nit: I think Euclidean should be capitalized, as it's derived from the mathematician's name.

sarahlwelton replied:
Fair! Fixed.

dgnicholson commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,49 @@
+= Use Vector Search for AI Applications
```
Simple, but great title. 🚀

dgnicholson commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,49 @@
+
+== About Vector Search
+
+Vector Search builds on Couchbase Capella's xref:search.adoc[Search Service] to provide vector index support.
```
Broken link
```suggestion
Vector Search builds on Couchbase Capella's xref:search:search.adoc[Search Service] to provide vector index support.
```

dgnicholson commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,49 @@
+[abstract]
+{description}
+
+== About Vector Search
```
This section is great. It's really hard to summarize these kinds of features w/o getting into the weeds, but you did it very well IMO.

dgnicholson commented on modules/vector-search/partials/nav.adoc:
```diff
@@ -0,0 +1,5 @@
+* xref:cloud:vector-search:vector-search.adoc[]
+** xref:cloud:vector-search:create-vector-search-index-ui.adoc[]
+** xref:cloud:vector-search:create-vector-search-index-rest-api.adoc[]
+** xref:cloud:vector-search:run-vector-search-ui.adoc[]
+** xref:cloud:vector-search:run-vector-search-rest-api.adoc[]
```
Out of curiosity, what was your reasoning on dividing the API instructions and the UI instructions into different pages and not do something like tabs?

sarahlwelton replied:
I wasn't sure if the API instructions were going to get too into the weeds, and honestly... It's just the complexity of the whole thing. 

I wanted to follow the structure I set up in basic search by having things separate. There was concerns about how hard search is to parse and understand, that I wanted to keep things dead-ass simple for the UI and then offer some help to get around the awkward API documentation with an explicit example. 

Tabs could be something to consider, but I also think there's something to be said for simplicity - especially with the differing prerequisites between UI and REST.

dgnicholson commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+[abstract]
+{description}
+
+Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Vector Search index and xref:customize-index.adoc#type-mappings[type mappings].
```
Broken links (I think it's supposed to be in the `:search:` directory?)

sarahlwelton replied:
Thought I caught all these from my copy/paste - thanks!

lamagnifica commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,49 @@
+
+== Using Vector Search Indexes
+
+Use a Capella database to store the data you want to use to provide context to a LLM. 
```
I'm finding the command structure of these first three sentneces odd. Maybe frame as, Using VSIs is a three-step process: 
1.
2.
3...

lamagnifica commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,49 @@
+
+Create a Vector Search index to use the vector embeddings from your data and find matches from a search query's generated vectors.
+
+If you do not want to use an integration like LangChain, you can also use the API for an existing LLM and one of their embedding models to generate these embeddings for your data.
```
Maybe reframe as a positive, "In addition to supporting integrations with frameworks like LangChain, you can also...

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+[abstract]
+{description}
+
+Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Vector Search index and xref:customize-index.adoc#type-mappings[type mappings].
```
incorrect links? both fail in the preview

lamagnifica replied:
which makes me wonder, does this quick mode topic also cover these advanced procedures? if not, maybe
After you create a VSI using quick made, you can configure more settings and add more advanced options for your Vector Search index and type mappings

dgnicholson commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+* You have a bucket with scopes and collections in your database. 
+For more information, see xref:cloud:clusters:data-service/manage-buckets.adoc[].
+
+* You have documents in a keyspace inside that bucket that contain vector embeddings.
```
Nit: I tripped over this line a little bit. I'm wondering if we change "that bucket" to "your bucket" would help.

```suggestion
* You have documents in a keyspace inside your bucket that contain vector embeddings.
```

dgnicholson commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+.. In the *Dimension* field, check that the value matches the total number of elements in your vector field's array. 
++
+The Search Service supports arrays up to 2048 elements.
+Capella should automatically fill in the dimension value for your selected array field.
```
Nit: perhaps make this a little less passive in tone—makes it sounds like it could break (I mean sure it can 😆 , but we don't want to convey that).

```suggestion
Capella automatically fills in the dimension value for your selected array field.
```

dgnicholson commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+For more information, see xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
+.. In the *Optimized for* list, select whether the Search Service should optimize Search queries for accuracy (*recall*) or speed (*latency*).
++
+For more information, see xref:xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
```
```suggestion
For more information, see xref:search:quick-index-field-options.adoc#quick-field[Quick Field Type Mapping Options].
```

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+. Expand the collection that contains these documents. 
+. In your document schema, select a field that contains your vector embeddings. 
++
+Vectors should be an array of floating point numbers.
```
maybe mention this in the prerequisites section?

lamagnifica replied:
also, is should the right word? are there workarounds if they are not floating point numbers, or in an array?

sarahlwelton replied:
No workaround - what would you suggest as an alternative?

lamagnifica replied:
Vectors must be represented as floating point numbers.

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+. In your document schema, select a field that contains your vector embeddings. 
++
+Vectors should be an array of floating point numbers.
+. Configure the options for your new child field as follows:
```
my what now? I missed where a new field got created. Maybe this is evident when actually using the UI, but perhaps state that "select a field that contains your vector embeddings" in the previous step has the result of adding a field to the schema ?

sarahlwelton replied:
I added a note to the previous step - let me know if that resolves your question?

lamagnifica replied:
I guess I still don't understand if there's a new field (child or otherwise) being added or if I'm just adding configuration to something I selected.

sarahlwelton replied:
So, you have the document schema displayed. You select the field from the JSON schema that contains the data you want to be able to search from your index. 

That's the child field, the field you selected. 

Then, you're adding configuration onto that. 

You select the field on one side, and configure on the other.

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+== Next Steps 
+
+This basic Vector Search index includes the vector embeddings from the field you specified in your type mapping.
+If you chose to add additional child fields, the Search Service can also return data from those fields when you run a Vector Search query. 
```
the references to child fields are quite confusing to me. Does optional step 11 add these child fields?

sarahlwelton replied:
Yes

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+
+== Next Steps 
+
+This basic Vector Search index includes the vector embeddings from the field you specified in your type mapping.
```
is the type mapping what I did in step 9? 
is this from the specified field or from the "child field" that got created?

sarahlwelton replied:
Yes, this is the child field you created a type mapping for.

dgnicholson commented on modules/vector-search/pages/create-vector-search-index-rest-api.adoc:
```diff
@@ -0,0 +1,59 @@
+
+== Prerequisites 
+
+* You have the Search Service enabled on a node in your database. 
+
+* You have a bucket with scopes and collections in your database. 
+
+* You have documents in a keyspace inside that bucket that contain vector embeddings. 
```
Nit: I wonder if the language of these prereq items should be reconciled more with the same ones in Quick Mode procedure.

sarahlwelton replied:
Good shout.

sarahlwelton committed:
[AV-68571] Consistency edits based on peer review

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+You can customize your Vector Search index like any other Search index to add additional data and improve search results. 
+For more information about how to customize an index, see xref:search:customize-index.adoc[].
+
+CAUTION: Some Search index features are only available in Advanced Mode. If you edit your Search index in Advanced Mode, you cannot return to Quick Mode and keep any Advanced Mode settings. 
```
instead of "return to Quick Mode" I'd rephrase more like, 
After you use Advanced Mode to configure your Search Index, you cannot use Quick Mode for any further edits. Using Quick Mode resets all Advanced Mode settings to their default state.
...If that is, in fact, what happens...

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-rest-api.adoc:
```diff
@@ -0,0 +1,59 @@
+
+To create a Search index with the REST API and Couchbase Capella: 
+
+. In your command-line tool, enter a `curl` command with the `XPUT` verb. 
```
verb --> method?

sarahlwelton replied:
Surprisingly they do refer to it as a "verb"

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-rest-api.adoc:
```diff
@@ -0,0 +1,59 @@
+
+. In your command-line tool, enter a `curl` command with the `XPUT` verb. 
+. Set your header content to include `Content-Type: application/json`.
+. Enter your username, password, and the Search Service endpoint on port `8094` with the name of the index you want to create: 
```
...with the ^fully-qualified^ name of the index...

sarahlwelton replied:
This depends on which endpoint you're using.

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-rest-api.adoc:
```diff
@@ -0,0 +1,59 @@
++
+TIP: If you remove the xref:search:search-index-params.adoc#uuid[uuid] and xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters, you can copy the Search index definition JSON payload from the Capella UI to use in a REST API call.
+For more information about how to create a Vector Search index with the UI, see xref:create-vector-search-index-ui.adoc[]. 
++
```
Maybe instead of indenting under the final step, make an  .Example or ==Example header to call more attention to this?

sarahlwelton replied:
I like this idea.

dgnicholson commented on modules/vector-search/pages/run-vector-search-ui.adoc:
```diff
@@ -0,0 +1,43 @@
+
+== Next Steps 
+
+If you do not get the search results you were expecting, you can change the xref:search-request-params.adoc[JSON payload for your Search query].
```
```suggestion
If you do not get the search results you were expecting, you can change the xref:search-request-params.adoc[JSON payload for your Search query].
```
```suggestion
If you do not get the search results you were expecting, you can change the xref:search:search-request-params.adoc[JSON payload for your Search query].
```

lamagnifica replied:
the links in preview are broken, so I opened the current Capella doc and found more info about Run A Simple Search with the Capella UI and Search Request JSON Properties, so never mind my confusion in this topic. Possibly just a reassuring sort of, Running a vector search index query in the Capella UI is similar to running a simple search (xref) for those of us who dropped in from another planet...

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-rest-api.adoc:
```diff
@@ -0,0 +1,59 @@
++
+Do not include the xref:search:search-index-params.adoc#uuid[uuid] or xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters. 
++
+TIP: If you remove the xref:search:search-index-params.adoc#uuid[uuid] and xref:search:search-index-params.adoc#sourceuuid[sourceUUID] parameters, you can copy the Search index definition JSON payload from the Capella UI to use in a REST API call.
```
I missed any mention of the JSON payload in the Quick Mode topic. Is that the same as creating it with the UI?
I feel like I'm missing 
- all the advanced setting / config info
- how to test/the results of setting up via the UI, presumably with a payload that can be reused via cURL

sarahlwelton replied:
Quick Mode does not expose you to the JSON payload at all, unless you want to view it. It's hidden. Quick Mode is a mode avilable in the UI. That's why the prereqs include "logging into the Couchbase Capella UI." 

Advanced settings are kept in another topic so users can just get the full breakdown of this huge JSON payload. 

Testing is just running a search and seeing if you get the results you want.

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,74 @@
+. Go to menu:Data Tools[Search].
+. Click btn:[Create Search Index].
++
+By default, *Quick Mode* should be on.
```
is this where you might switch to Advanced Mode?

sarahlwelton replied:
Yes

dgnicholson reviewed (changes requested):
This is a monster, but really well done @sarahlwelton. 🚀 

I focused mostly on structure, functionality, and readability and my only main concerns are around several broken links, but otherwise I think this is good to go. Anything with "Nit" is at your discretion.

lamagnifica commented on modules/vector-search/pages/run-vector-search-ui.adoc:
```diff
@@ -0,0 +1,43 @@
+= Run A Vector Search with the Capella UI 
```
lc A

lamagnifica commented on modules/vector-search/pages/run-vector-search-ui.adoc:
```diff
@@ -0,0 +1,43 @@
+= Run A Vector Search with the Capella UI 
+:page-topic-type: guide
+:description: Run a Vector Search query from the Couchbase Capella UI to preview the search results from a Vector Search index.
```
is this a verification/testing step? that is, why would I want to preview?

sarahlwelton replied:
"preview and test"?

lamagnifica commented on modules/vector-search/pages/run-vector-search-ui.adoc:
```diff
@@ -0,0 +1,43 @@
+[source,json]
+----
+include::example$run-vector-search-payload-ui.jsonc[]
+----
```
I haven't seen the UI, but this strikes me as very different from other queries in Capella. 
I would expect first a SQL++-looking query, and then the JSON result, rather than both together in one JSON -formatted document
Just raising that as possibly something to state explicitly, that these queries are not SQL++ queries, and whatever steps they may need to take to embed the query in JSON formatting (and when to expect the result to get added to the JSON)

sarahlwelton replied:
Because it is very different from all other queries in Capella. This isn't based on SQL++ at all - it's full text search. 

This also isn't the result, but the actual query you need to put into the Search field to return a result.

lamagnifica commented on modules/vector-search/pages/run-vector-search-ui.adoc:
```diff
@@ -0,0 +1,43 @@
++
+By using the special `match_none` query in the `query` field, the Search query only returns the `k` number of similar vectors.
+
+. Press kbd:[Enter].
```
[EDIT disregard, see next comment]
...and now I'm not sure what I was accomplishing by doing these steps. I assumed this was a test, a way to see if what you set up was enabled and giving responses to queries, but now it's almost like you're training it?
Is there no response to the Enter?

lamagnifica commented on modules/vector-search/pages/create-vector-search-index-rest-api.adoc:
```diff
@@ -0,0 +1,59 @@
+= Create a Vector Search Index with the REST API and curl/HTTP
```
I've always used cURL -- not a cb standard?

sarahlwelton replied:
We had a whole discussion about this when I first started doing this and apparently even the guy who created curl doesn't write it as cURL anymore? Who knew

sarahlwelton committed:
Update modules/n1ql/pages/n1ql-language-reference/searchfun.adoc

Co-authored-by: Daniel Nicholson <dgnicholson@users.noreply.github.com>

lamagnifica reviewed (commented):
Coming in as a completely naive user (no vector search background OR Capella search background), there were a few confusing points, though for the most part just a few nits and suggestions for rephrasing. Two thumbs up!

sarahlwelton committed:
Update modules/vector-search/pages/vector-search.adoc

Co-authored-by: Daniel Nicholson <dgnicholson@users.noreply.github.com>

sarahlwelton committed:
[AV-68571] Cleaning up some comments from peer review

sarahlwelton committed:
Merge branch 'AV-68571-vector-search-7-6-capella' of https://github.com/couchbaselabs/docs-devex into AV-68571-vector-search-7-6-capella

sarahlwelton committed:
[AV-68571] Undoing a mistaken change.

sarahlwelton committed:
[AV-68571] Applying some changes that came in through peer review.

sarahlwelton committed:
[AV-68571] Another correction from peer review comments

sarahlwelton committed:
[AV-68571] Adding a tip to make the connection between Vector Search and Search clear.

ElliotFrancisHunter commented on modules/n1ql/pages/n1ql-language-reference/searchfun.adoc:
```diff
@@ -1,54 +1,54 @@
 To use any of the search functions, the Search service must be available on the cluster.
-It is also recommended, but not required, that you should create suitable full text indexes for the searches that you need to perform.
-For more information, refer to xref:server:fts:fts-quickstart-guide.adoc#preparing-for-full-text-searches[Preparing for Full Text Searches].
+It is also recommended, but not required, that you should create suitable Search indexes for the searches that you need to perform.
```
This feels a little awkward on the scan, can we cut the 'but not required' out and still carry the intent across?

sarahlwelton replied:
I didn't write this but I'm sure we could make that change.

ElliotFrancisHunter commented on modules/search/pages/search.adoc:
```diff
@@ -52,9 +52,19 @@ You can run a Search query:
 * xref:customize-index.adoc[]
 * xref:index-aliases.adoc[]
 * xref:run-searches.adoc[]
+* xref:vector-search:vector-search.adoc[]
```
Was it intentional to have two links to the same page close to each other? I understand the usefulness, just confirming this was deliberate :)

sarahlwelton replied:
Yup, deliberate! Just in case people skip on by without reading.

ElliotFrancisHunter commented on modules/search/pages/customize-index.adoc:
```diff
@@ -1,51 +1,69 @@
-= Customize a Search Index with the Capella UI
+= Search Index Features
```
Seconding Daniel's thoughts, succinct and useful titles throughout. Awesome stuff!

ElliotFrancisHunter commented on modules/search/pages/quick-index-field-options.adoc:
```diff
@@ -0,0 +1,129 @@
+
+You can create the following types of type mappings in Quick Mode: 
+
+* <<quick-collection,>>
```
Remove commas

sarahlwelton replied:
It's part of the markup, I'm going to leave that in.

ElliotFrancisHunter reviewed (commented):
This is some really awesome work! Other than a handful of queries there wasn't much I could find issue-wise. Especially impressive as this PR is humungous 😱

bfavini commented on modules/search/pages/create-quick-index.adoc:
```diff
@@ -0,0 +1,56 @@
+====
+Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).
+
+For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
```
Should probably say "For Couchbase Server 7.6 and later"

sarahlwelton replied:
Even though this is Capella? I'm torn on how to refer to the version.

bfavini commented on modules/vector-search/pages/vector-search.adoc:
```diff
@@ -0,0 +1,50 @@
+Vector Search builds on Couchbase Capella's xref:search:search.adoc[Search Service] to provide vector index support.
+You can use these new Vector Search indexes for Retrieval Augmented Generation (RAG) with an existing Large Language Model (LLM). 
+
+Using Vector Search, Capella, an embedding model, and your chosen LLM, you can develop AI applications while giving context and up-to-date information from your own data.
```
Agree we don't want to get into the weeds...but it feels like we could use a bit more context around RAG, LLM, and embedding models. I would not hold up GA for this, but I think some simple definitions of these things and the role they play in AI would be helpful, particularly to new developers

bfavini commented on modules/search/pages/create-quick-index.adoc:
```diff
@@ -0,0 +1,56 @@
+[abstract]
+{description}
+
+Use xref:create-search-index-ui.adoc[Advanced Mode] to add additional settings and more advanced options for your Search index and xref:customize-index.adoc#type-mappings[type mappings].
```
Would be even more people to help people understand what exactly they're giving up when they choose quick mode or when it's the right choice.....(Choose Quick Mode when....)

bfavini commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,78 @@
+= Create a Vector Search Index in Quick Mode
+:page-topic-type: guide
+:description: Use Quick Mode to create a Vector Search index in Couchbase Capella.
+
```
Similar comment in the general FTS section--would be helpful to call out what Quick Mode doesn't give me...why/when would I choose Quick Mode over Advanced and vice versa?

bfavini commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,78 @@
+====
+Your index name must start with an alphabetic character (a-z or A-Z). It can only contain alphanumeric characters (a-z, A-Z, or 0-9), hyphens (-), or underscores (_).
+
+For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
```
Couchbase Server version7.6 or later

bfavini commented on modules/vector-search/pages/create-vector-search-index-ui.adoc:
```diff
@@ -0,0 +1,78 @@
+
+For version 7.6 and later, your index name must be unique inside your selected bucket and scope. You cannot have 2 indexes with the same name inside the same bucket and scope.
+====
+. Under *Type Mappings*, in the *Bucket* list, select the bucket that contains the documents you want to include in your index. 
```
I originally intended to recommend using a different structure here, removing some of the conceptual or reference information from the steps. In the spirit of NOT documenting how to use the UI.

However, having read and reread this several times, I don't think you can avoid it. The procedure in the UI is just too complex/tricky.

bfavini reviewed (approved):
Nice job on this massive PR. Some general comments to enhance moving forward. Nothing should hold up GA.

sarahlwelton committed:
[AV-68571] Applying more suggestions from peer review

sarahlwelton committed:
[AV-68571] Feedback from SME review

sarahlwelton committed:
[AV-68571] More updates from SME review

sarahlwelton committed:
[AV-68571] More SME review updates - adding procedure for running a vector search with Python SDK

sarahlwelton committed:
[AV-68571] Copying some changes over for consistency.

sarahlwelton committed:
[AV-68571] Quick consistency edit.

sarahlwelton committed:
[AV-68571] Consistency edits and incorporating comments from another SME

sarahlwelton committed:
[AV-68571] Adding clarification based on MB-60141

sarahlwelton committed:
[AV-68571] Big consistency pass in tandem with branch on Server to clear up any remaining inconsistencies between files.

sarahlwelton committed:
[AV-68965] Updating integrations page for LangChain

Per Richard Smedley, copying over missed update to add in Terraform as well.

sarahlwelton committed:
[AV-68571] Adding caution to all vector search creation pages based on feedback from SRE

sarahlwelton committed:
[AV-68571] Adding attributes to handle product and UI names to future-proof updates.

Adding proper expansion formatting to variables in REST API paths/curl commands

sarahlwelton committed:
[AV-68571] May -> Can, per peer review

sarahlwelton committed:
[AV-68571] Updating integrations.adoc

sarahlwelton committed:
[AV-68571] Fixing broken links, adding missing file.

sarahlwelton committed:
[AV-68571] Changing integrations.adoc into a partial to pull in from Server

sarahlwelton committed:
[AV-68571] Fix some screwy attributes, add latest samples for SDK code and rename topic to be the one-stop for SDK and Vector Search

sarahlwelton committed:
[AV-68571] Remove caution about large index sizes.
